src/librustc/ty/item_path.rs

```diff
@@ -189,10 +182,15 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
                     Some(ExternCrate {
                         src: ExternCrateSource::Extern(def_id),
                         direct: true,
+                        span,
                         ..
                     }) => {
                         debug!("try_push_visible_item_path: def_id={:?}", def_id);
-                        self.push_item_path(buffer, def_id, pushed_prelude_crate);
+                        if !span.is_dummy() {
```

Comment:
Similarly, comment here -- or maybe we can pull this into some kind of common helper function with the code above?

Reply:
So in a different branch I was able to deduplicate this by moving the local mode logic from `push_krate_path` to `try_push_visible_item_path` and relying on the fact that `try_push_visible_item_path` can override the behavior of `push_item_path`. But maybe we can avoid the dummy span check here by some other means.